File: leetcode/algorithms/643_maximum_average_subarray_1/main.java
class MaximumAverageSubarrayI {
    public double findMaxAverage(int[] nums, int k) {
        double currSum = 0, maxSum = 0;

        for (int i = 0; i < k; i++) {
            currSum += nums[i];
        }
        maxSum = currSum;

        for (int i = k; i < nums.length; i++) {
            currSum += nums[i] - nums[i - k];
            maxSum = Math.max(currSum, maxSum);
        }

        return maxSum / k;
    }

    // Solution
    public double solution(int[] nums, int k) {
        long sum = 0;

        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }

        long max = sum;

        for (int i = k; i < nums.length; i++) {
            sum += nums[i] - nums[i - k];
            max = Math.max(max, sum);
        }

        return max / 1.0 / k;
    }
}

File: leetcode/algorithms/643_maximum_average_subarray_1/main.py
from typing import List


class MaximumAverageSubarrayI:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        currSum: float = 0
        maxSum: float = 0

        for i in range(k):
            currSum += nums[i]

        maxSum = currSum

        for i in range(k, len(nums)):
            currSum += nums[i] - nums[i - k]
            maxSum = max(currSum, maxSum)

        return maxSum / k

    # Solution
    def solution(self, nums: List[int], k: int) -> float:
        # Initialize currSum and maxSum to the sum of the initial k elements
        currSum = maxSum = sum(nums[:k])

        # Start the loop from the kth element
        # Iterate until you reach the end
        for i in range(k, len(nums)):
            # Subtract the left element of the window
            # Add the right element of the window
            currSum += nums[i] - nums[i - k]

            # Update the max
            maxSum = max(maxSum, currSum)

        # Since the problem requires average, we return the average
        return maxSum / k
